sklearn/preprocessing/_encoders.py

```diff
@@ -692,12 +693,14 @@ def get_feature_names(self, input_features=None):
 
         feature_names = []
         for i in range(len(cats)):
-            names = [input_features[i] + "_" + str(t) for t in cats[i]]
+            names = [str(input_features[i]) + "_" + str(t) for t in cats[i]]
```

Comment:
This is another big issue, where we have not decided what to do. In https://github.com/scikit-learn/scikit-learn/pull/21754#pullrequestreview-813973424 we elected to error and give a better error message.

I wouldn't include this fix in this PR as it increases the scope and focus on the `None` issue.